src/build_workflow/build_artifact_check.py

```diff
@@ -7,16 +7,18 @@
 import os
 from abc import ABC, abstractmethod
 
+from build_workflow.build_target import BuildTarget
+
 
 class BuildArtifactCheck(ABC):
     class BuildArtifactInvalidError(Exception):
-        def __init__(self, path, message):
+        def __init__(self, path: str, message: str) -> None:
             self.path = path
             super().__init__(f"Artifact {os.path.basename(path)} is invalid. {message}")
 
-    def __init__(self, target):
+    def __init__(self, target: 'BuildTarget') -> None:
```

Comment:
Are the quotes needed?

Reply:
There were places where the 'string' version of the type name was needed, this was because the type was defined later in the file to prevent the circular reference problems.  It should not be needed elsewhere unless it serves another purpose?

Reply:
I see. Then the quotes are not needed. Will fix it. Thank you for pointing it out.